src/mds/CInode.h

```diff
@@ -77,24 +78,63 @@ extern int num_cinode_locks;
 class InodeStoreBase {
 public:
   typedef inode_t<mempool::mds_co::pool_allocator> mempool_inode;
+  struct refcounted_inode : public mempool_inode, public RefCountedObject {
+    MEMPOOL_CLASS_HELPERS();
+    refcounted_inode() {}
+    refcounted_inode(const refcounted_inode& o) : mempool_inode(o) {}
+  };
+  typedef boost::intrusive_ptr<refcounted_inode> inode_ptr;
```

Comment:
Why not `std::shared_ptr`?

Reply:
Also, we should prefer `using inode_ptr = ...`.

Reply:
because we can't specify custom allocater for std::make_shared

Reply:
how about `std::allocate_shard<>`? see https://en.cppreference.com/w/cpp/memory/shared_ptr/allocate_shared

Reply:
converted to shared_ptr